services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -274,39 +277,28 @@ public Builder filterExpression(Expression filterExpression) {
         }
 
         /**
-         * <p>
-         * A string that identifies one or more attributes to retrieve from the table. These attributes can include
-         * scalars, sets, or elements of a JSON document. The attributes in the expression must be separated by commas.
-         * </p>
-         * <p>
-         * If no attribute names are specified, then all attributes will be returned. If any of the requested attributes
-         * are not found, they will not appear in the result.
-         * </p>
-         * <p>
-         * For more information, see <a href=
-         * "https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.AccessingItemAttributes.html"
-         * >Accessing Item Attributes</a> in the <i>Amazon DynamoDB Developer Guide</i>.
-         * </p>
+         * The list of the attributes's paths to be retrieved from the database.
          *
-         * @param projectionExpression
-         *        A string that identifies one or more attributes to retrieve from the table. These attributes can
-         *        include scalars, sets, or elements of a JSON document. The attributes in the expression must be
-         *        separated by commas.</p>
-         *        <p>
-         *        If no attribute names are specified, then all attributes will be returned. If any of the requested
-         *        attributes are not found, they will not appear in the result.
-         *        </p>
-         *        <p>
-         *        For more information, see <a href=
-         *        "https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.AccessingItemAttributes.html"
-         *        >Accessing Item Attributes</a> in the <i>Amazon DynamoDB Developer Guide</i>.
+         * @param attributesToProject
+         *        The list of the attributes's paths to be retrieved from the database.
          * @return Returns a reference to this object so that method calls can be chained together.
          */
-        public Builder projectionExpression(String projectionExpression) {
-            this.projectionExpression = projectionExpression;
+        public Builder attributesToProject(List<String> attributesToProject) {
+            this.attributesToProject.addAll(attributesToProject);
```

Comment:
We have a particular style around collections in builders that we should try and consistently follow. For an optional list like this:
1. If no values are given to the builder then the builder should pass 'null' to the constructor to indicate 'no optional values supplied'.
2. On instantiation, the object should shallow copy all the values out of the builder array and store it in an immutable List implementation (we typically use Collections.unmodifiableList of an ArrayList).
3. On the builder we provide a method to add a single element to the existing builder list (or create one if it is still null), these are named 'addXXX(...)' where XXX is the singular form of the collection noun. (eg: addAttributeToProject(...)).
4. Also on the builder we provide 2 overloaded methods to replace the entire list with one that is supplied by taking a shallow copy of it (this will overwrite any values already stored in the builder list). These are named 'XXX(...)' where XXX is the plural form of the collection noun. Eg: (attributesToProject(...)). The overload provides a varargs form and a List<...> form.